lib/buffer.js

```diff
@@ -371,7 +371,7 @@ Buffer.prototype.toString = function() {
     var result = slowToString.apply(this, arguments);
   }
   if (result === undefined)
-    throw new Error('toString failed');
+    throw new Error('toString failed - buffer larger than maximum string size ' + binding.kStringMaxLength);
```

Comment:
Nit: Error messages everywhere else in `lib` use `:` to separate the executive summary from the gory details but this one is using `-`. Switch to `:`.